### ISSUE TITLE
declare_directory outputs with subdirectories create execroots that cannot be cleaned with --async

### ISSUE THREAD

**Issue by werkt** [bazelbuild/bazel#7382]
### Description of the problem / feature request:

`bazel clean --async` cannot clean directory trees with plural depth created via `ctx.declare_directory`. This state is irreparable without `bazel clean --expunge` and non-idempotent - subsequent `bazel clean --async` invocations error on the execroot directory rename with `(Directory not empty)`. `bazel clean --expunge --async` demonstrates similar behavior, and will leave a temp directory under the user bazel cache root.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. Populate the following into a workspace:

BUILD
```
load(":create_directory.bzl", "create_directory")

create_directory(name="foo")
```
create_directory.bzl
```
def _impl(ctx):
  out = ctx.actions.declare_directory(ctx.attr.name + ".d")

  runfiles = ctx.runfiles(files = [out])
  script = "mkdir {out}/{name}".format(out = out.path, name = ctx.attr.name)
  ctx.actions.run_shell(
      outputs = [out],
      command = script)

  return [DefaultInfo(
      files = depset([out]),
      runfiles = runfiles,
  )]

create_directory = rule(
    implementation = _impl,
)
```

1. run `bazel build :foo` (should succeed)
1. run `bazel clean --async` (should succeed)
1. copy the directory name after `Output tree moved to ` in the output of `clean --async`
1. run `stat <directory from ^^>` (should succeed, be a directory, not having been deleted)
1. run `bazel build :foo && bazel clean --async` (should fail with `(Directory not empty)`) error

Only `--expunge` successfully removes that path, and repeating the process above with `--expunge --async` leaves full `output_base` tmpdir copies that require `chmod`s for all `declare_directory` trees to get rid of.

### What operating system are you running Bazel on?

Ubuntu 18.04

### What's the output of `bazel info release`?

release 0.22.0

**Comment by JaredNeil** [comment]
Looks like this was fixed in response to #12125. Not sure how I didn't find this when I was searching for issues before creating that one.

**Comment by sgowroji** [comment]
Hi there! We're doing a clean up of old issues and will be closing this one. Please reopen if you’d like to discuss anything further. We’ll respond as soon as we have the bandwidth/resources to do so.
